### ISSUE TITLE
problem occurred configuring project ':isar_flutter_libs'.

### ISSUE THREAD

**Issue by angelru** [isar/isar#1470]
```
flutter build apk                                                               


FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring project ':isar_flutter_libs'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.
```

```
buildscript {
    ext.kotlin_version = '1.9.10'
    repositories {
        google()
        mavenCentral()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:8.1.2'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
    }
```
}

**Comment by Yongle-Fu** [comment]
same issue
and downgrade to com.android.tools.build:gradle:7.4.2 worked

**Comment by angelru** [comment]
Yes, but I get this:
Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

I need the grandle 8

**Comment by 988494** [comment]
com.android.tools.build:gradle:8.1.2 在中国地区加速访问仓库中 ，没有这个版本，是否可以降低为8.1.1呢？

**Comment by angelru** [comment]
not working

**Comment by Peng1104** [comment]
Its a problem of Isar, It needs gradle 8.0.0+ support I hope that they add this soon

**Comment by angelru** [comment]
I imagine there is no alternative solution except use version 7

**Comment by Phero49** [comment]
same issue here  my project requires 8.0+ and if i use  7xx am running into errors on build  . has one figured out how to fix this without downgrading

**Comment by Hieu86355** [comment]
Hey guys, any updates on this?

**Comment by Phero49** [comment]
maybe rebuilding the package would work

**Comment by asaarnak** [comment]
Seems like this has been fixed in isar v4 which is in dev.
https://github.com/isar/isar/blame/main/packages/isar_flutter_libs/android/build.gradle#L25-L27

**Comment by gilsonjuniorpro** [comment]
any idea when a version of isar_generator with namespaces will be released?

**Comment by nobkd** [comment]
For Isar 4, the generator is built in: https://github.com/isar/isar/issues/1357#issuecomment-1659027403

**Comment by MrLittleWhite** [comment]
same issue on lastest release version [3.1.0+1](https://pub.dev/packages/isar_flutter_libs/versions/3.1.0+1)， any solution？

**Comment by outdoorapps** [comment]
> same issue on lastest release version [3.1.0+1](https://pub.dev/packages/isar_flutter_libs/versions/3.1.0+1)， any solution？

Downgrading to gradle 7.4.2 seems to be the only workaround right now until they release Isar v4 (which you might have noticed the project haven't been updated since Aug 2023).

**Comment by MrLittleWhite** [comment]
I figure it out.  It can be fixed by modify the cache of isar_flutter_libs pub like this https://github.com/isar/isar/blame/main/packages/isar_flutter_libs/android/build.gradle#L25-L27. 
Or you can fix by https://github.com/MrLittleWhite/isar_flutter_libs.

**Comment by ssthiago** [comment]
> I figure it out. It can be fixed by modify the cache of isar_flutter_libs pub like this https://github.com/isar/isar/blame/main/packages/isar_flutter_libs/android/build.gradle#L25-L27. Or you can fix by https://github.com/MrLittleWhite/isar_flutter_libs.

Perfect! Thanks @MrLittleWhite

**Comment by vicenterusso** [comment]
Also you could try using this fork

https://github.com/isar-community/isar

We are keeping the same isar base code but with dependencies updated, and eventually new features

**Comment by angelru** [comment]
Isn't it going to be solved by Isar?

**Comment by Hesamedin** [comment]
Same issue!

**Comment by tech-dhruv** [comment]
Same issue!

**Comment by deam91** [comment]
Same issue! When you plan to release Isar 4 ?

**Comment by XababaDalabama** [comment]
Any news about this subject ? I think Android Gradle Plugin is version 8.6.1
And I also think some other dependencies can't be upgraded.
Flutter (Channel stable, 3.24.3 / Android Studio (version 2024.1)

**Comment by robertoalvarezjunior** [comment]
I solved this problem with the following commands:
dart pub cache clean
flutter clean
flutter pub get

**Comment by andrepurnomo** [comment]
Try this solution https://stackoverflow.com/a/79027489

My problem solved with add this code to android/build.gradle
Flutter 3.24.3
isar_flutter_libs: ^3.1.0+1
```
subprojects {
        afterEvaluate { project ->
            if (project.hasProperty('android')) {
                project.android {
                    if (namespace == null) {
                        namespace project.group
                    }
               }
           }
      }
   }
```

**Comment by amithapa** [comment]
> Try this solution https://stackoverflow.com/a/79027489
> 
> My problem solved with add this code to android/build.gradle Flutter 3.24.3 isar_flutter_libs: ^3.1.0+1
> 
> ```
> subprojects {
>         afterEvaluate { project ->
>             if (project.hasProperty('android')) {
>                 project.android {
>                     if (namespace == null) {
>                         namespace project.group
>                     }
>                }
>            }
>       }
>    }
> ```

Works, thank you @andrepurnomo

**Comment by Matzu89** [comment]
TLDR;

Did the trick for me. Did not need to update gradle. 

> Also you could try using this fork
> 
> https://github.com/isar-community/isar
> 
> We are keeping the same isar base code but with dependencies updated, and eventually new features

**Comment by solinkp** [comment]
If the above solutions don't work (like my case)
here's an alternative that might apply to your case:
**I'm using:**
Flutter 3.24.5
gradle-8.10.2
isar_flutter_libs: ^3.1.0+1

Added these 2 overrides to pubspec
```
dependency_overrides:
  analyzer: 6.7.0
  custom_lint_core: 0.6.5
```

**Comment by mrmhthegreat** [comment]
Locate Your Package dir and put replace build gradle file

android {
    namespace "dev.isar.isar_flutter_libs"
    compileSdkVersion 30

    defaultConfig {
        minSdkVersion 16
    }
}

**Comment by Frankdroid7** [comment]
> Try this solution https://stackoverflow.com/a/79027489
> 
> My problem solved with add this code to android/build.gradle Flutter 3.24.3 isar_flutter_libs: ^3.1.0+1
> 
> ```
> subprojects {
>         afterEvaluate { project ->
>             if (project.hasProperty('android')) {
>                 project.android {
>                     if (namespace == null) {
>                         namespace project.group
>                     }
>                }
>            }
>       }
>    }
> ```

This works. Thanks!

**Comment by 0xharkirat** [comment]
> I figure it out. It can be fixed by modify the cache of isar_flutter_libs pub like this https://github.com/isar/isar/blame/main/packages/isar_flutter_libs/android/build.gradle#L25-L27. Or you can fix by https://github.com/MrLittleWhite/isar_flutter_libs.

This is still working. Anyone facing this issue can use this.

**Comment by baiyulong** [comment]
Change version to isar: ^4.0.0-dev.14
It's working for me.

**Comment by oyeasif** [comment]
I just solve with gradle 7.4.2

**Comment by kennyyy5** [comment]
> Change version to isar: ^4.0.0-dev.14 It's working for me.

when i did this, it showed me errors of dependency, did you also change the version of isar_flutter_libs/isar_generator?

**Comment by oyeasif** [comment]
> > Change version to isar: ^4.0.0-dev.14 It's working for me.
> 
> when i did this, it showed me errors of dependency, did you also change the version of isar_flutter_libs/isar_generator?

Dont change the version you can try these two methods.

1. subprojects {
        afterEvaluate { project ->
            if (project.hasProperty('android')) {
                project.android {
                    if (namespace == null) {
                        namespace project.group
                    }
               }
           }
      }
   }
   Add this into your android/build.gradle file 
  
   OR
   
   2. change     id "com.android.application" version "8.1.0"  to 7.4.2 in settings.gradle file.

**Comment by rafsanza-hub** [comment]
> Try this solution https://stackoverflow.com/a/79027489
> 
> My problem solved with add this code to android/build.gradle Flutter 3.24.3 isar_flutter_libs: ^3.1.0+1
> 
> ```
> subprojects {
>         afterEvaluate { project ->
>             if (project.hasProperty('android')) {
>                 project.android {
>                     if (namespace == null) {
>                         namespace project.group
>                     }
>                }
>            }
>       }
>    }
> ```



> Try this solution https://stackoverflow.com/a/79027489
> 
> My problem solved with add this code to android/build.gradle Flutter 3.24.3 isar_flutter_libs: ^3.1.0+1
> 
> ```
> subprojects {
>         afterEvaluate { project ->
>             if (project.hasProperty('android')) {
>                 project.android {
>                     if (namespace == null) {
>                         namespace project.group
>                     }
>                }
>            }
>       }
>    }
> ```

**Comment by rafsanza-hub** [comment]
this works, thank you

**Comment by rafsanza-hub** [comment]
> Try this solution https://stackoverflow.com/a/79027489
> 
> My problem solved with add this code to android/build.gradle Flutter 3.24.3 isar_flutter_libs: ^3.1.0+1
> 
> ```
> subprojects {
>         afterEvaluate { project ->
>             if (project.hasProperty('android')) {
>                 project.android {
>                     if (namespace == null) {
>                         namespace project.group
>                     }
>                }
>            }
>       }
>    }
> ```

this works, thanks

**Comment by abdulbosit973o** [comment]
allprojects {
    repositories {
        google()
        mavenCentral()
    }
    rootProject.buildDir = "../build"
    subprojects {
        afterEvaluate { project ->
            if (project.plugins.hasPlugin("com.android.application") ||
                    project.plugins.hasPlugin("com.android.library")) {
                project.android {
                    compileSdkVersion 34
                    buildToolsVersion "34.0.0"
                }
            }
            if (project.hasProperty("android")) {
                project.android {
                    if (namespace == null) {
                        namespace project.group
                    }
                }
            }
        }
        // ============
        project.buildDir = "${rootProject.buildDir}/${project.name}"
        project.evaluationDependsOn(':app') // add this also
    }
    subprojects {
        project.evaluationDependsOn(":app")
    }
}

rootProject.buildDir = "../build"
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(":app")
}

tasks.register("clean", Delete) {
    delete rootProject.buildDir
}


i change my build.gradle to this code and it is working

**Comment by hafidzaini** [comment]
> > Try this solution https://stackoverflow.com/a/79027489
> > My problem solved with add this code to android/build.gradle Flutter 3.24.3 isar_flutter_libs: ^3.1.0+1
> > ```
> > subprojects {
> >         afterEvaluate { project ->
> >             if (project.hasProperty('android')) {
> >                 project.android {
> >                     if (namespace == null) {
> >                         namespace project.group
> >                     }
> >                }
> >            }
> >       }
> >    }
> > ```
> 
> this works, thanks

thanks 

> allprojects { repositories { google() mavenCentral() } rootProject.buildDir = "../build" subprojects { afterEvaluate { project -> if (project.plugins.hasPlugin("com.android.application") || project.plugins.hasPlugin("com.android.library")) { project.android { compileSdkVersion 34 buildToolsVersion "34.0.0" } } if (project.hasProperty("android")) { project.android { if (namespace == null) { namespace project.group } } } } // ============ project.buildDir = "${rootProject.buildDir}/${project.name}" project.evaluationDependsOn(':app') // add this also } subprojects { project.evaluationDependsOn(":app") } }
> 
> rootProject.buildDir = "../build" subprojects { project.buildDir = "${rootProject.buildDir}/${project.name}" } subprojects { project.evaluationDependsOn(":app") }
> 
> tasks.register("clean", Delete) { delete rootProject.buildDir }
> 
> i change my build.gradle to this code and it is working

thank you

been stuck with this for hours.
finally its resolve

**Comment by nathaponb** [comment]
> Try this solution https://stackoverflow.com/a/79027489
> 
> My problem solved with add this code to android/build.gradle Flutter 3.24.3 isar_flutter_libs: ^3.1.0+1
> 
> ```
> subprojects {
>         afterEvaluate { project ->
>             if (project.hasProperty('android')) {
>                 project.android {
>                     if (namespace == null) {
>                         namespace project.group
>                     }
>                }
>            }
>       }
>    }
> ```

for build.gradle.kts 👇

```kts
subprojects {
    afterEvaluate {
        if (project.hasProperty("android")) {
            val androidExtension = project.extensions.findByName("android")
            if (androidExtension is com.android.build.gradle.BaseExtension) {
                if (androidExtension.namespace == null) {
                    androidExtension.namespace = project.group.toString()
                }
            }
        }
    }
}

// before this block
subprojects {
    project.evaluationDependsOn(":app")
}

```

**Comment by RaphCayou** [comment]
Thanks @nathaponb! I am not so familiar with this and I was only seeing the Groovy versions and not the Kotlin .kts for the gradle file. Worked for me!

**Comment by Muhammad-Radwan** [comment]
android/settings.gradle
plugins section
change "com.android.application" version to 8.2.1
this was suggested by android studio, it worked for me

**Comment by llYakarll** [comment]
**Fix for `isar_flutter_libs` namespace error in `android/build.gradle.kts`**

If you get this error:
```
Namespace not specified. Specify a namespace in the module's build file
```
or
```
Cannot run Project.afterEvaluate(Action) when the project is already evaluated
```

Replace the **entire contents** of `android/build.gradle.kts` with the following:

```kotlin
allprojects {
    repositories {
        google()
        mavenCentral()
    }
}

val newBuildDir: Directory =
    rootProject.layout.buildDirectory
        .dir("../../build")
        .get()
rootProject.layout.buildDirectory.value(newBuildDir)

subprojects {
    val newSubprojectBuildDir: Directory = newBuildDir.dir(project.name)
    project.layout.buildDirectory.value(newSubprojectBuildDir)
}

subprojects {
    afterEvaluate {
        if (project.hasProperty("android")) {
            val androidExtension = project.extensions.findByName("android")
            if (androidExtension is com.android.build.gradle.BaseExtension) {
                if (androidExtension.namespace == null) {
                    androidExtension.namespace = project.group.toString()
                }
            }
        }
    }
}

subprojects {
    project.evaluationDependsOn(":app")
}

tasks.register<Delete>("clean") {
    delete(rootProject.layout.buildDirectory)
}
```

**What this does:** Automatically assigns a namespace to Flutter plugins (like `isar_flutter_libs`) that don't declare one. Required for newer Android Gradle Plugin versions.
